metafacture-io/src/test/java/org/metafacture/io/HttpOpenerTest.java

```diff
@@ -0,0 +1,316 @@
+        shouldPerformRequest(TEST_URL, HttpOpener.Method.GET, (o, u) -> {
+            o.setHeader("x-api-key: " + TEST_STRING);
+            o.setHeader("x-other-header: " + TEST_STRING);
+        }, "x-api-key", "x-other-header");
+    }
+
+    @Test
+    public void shouldPerformPostRequestWithContentTypeParameter() throws IOException {
+        shouldPerformRequest(REQUEST_BODY, HttpOpener.Method.POST, (o, u) -> {
+            o.setMethod(HttpOpener.Method.POST);
+            o.setUrl(u);
+            o.setContentType(TEST_STRING);
+        }, "Content-Type");
+    }
+
+    @Test
+    public void shouldPerformPostRequestWithEncodingParameter() throws IOException {
+        final String encoding = "ISO-8859-1";
+        final String header = "Accept-Charset";
+        final StringValuePattern value = WireMock.equalTo(encoding);
+
+        try {
+            shouldPerformRequest(REQUEST_BODY, HttpOpener.Method.POST, (o, u) -> {
+                o.setMethod(HttpOpener.Method.POST);
+                o.setUrl(u);
+                o.setEncoding(encoding);
+            }, s -> s.withHeader(header, value), q -> q.withHeader(header, value), null);
+        }
+        catch (final ComparisonFailure e) {
+            Assert.assertEquals("expected:<response b[ö]dy> but was:<response b[Ã¶]dy>", e.getMessage());
+        }
+    }
+
+    @Test
+    public void shouldPerformPostRequestWithEncodingParameterAndContentEncodingResponseHeader() throws IOException {
+        final String encoding = "ISO-8859-1";
+        final String header = "Accept-Charset";
+        final StringValuePattern value = WireMock.equalTo(encoding);
+
+        shouldPerformRequest(REQUEST_BODY, HttpOpener.Method.POST, (o, u) -> {
+                o.setMethod(HttpOpener.Method.POST);
+                o.setUrl(u);
+                o.setEncoding(encoding);
+            },
+            s -> s.withHeader(header, value),
+            q -> q.withHeader(header, value),
+            r -> r.withHeader("Content-Encoding", "UTF-8")
+        );
+    }
+
+    @Test
+    public void shouldPerformGetRequestWithErrorResponse() throws IOException {
+        shouldPerformRequest(TEST_URL, HttpOpener.Method.GET, (o, u) -> {},
+                null, null, WireMock.badRequest().withBody(RESPONSE_BODY), "ERROR: " + TEST_ERROR);
+    }
+
+    @Test
+    public void shouldPerformGetRequestWithErrorResponseAndErrorPrefixParameter() throws IOException {
+        shouldPerformRequest(TEST_URL, HttpOpener.Method.GET, (o, u) -> o.setErrorPrefix(TEST_STRING),
+                null, null, WireMock.badRequest().withBody(RESPONSE_BODY), TEST_STRING + TEST_ERROR);
+    }
+
+    @Test
+    public void shouldPerformGetRequestWithErrorResponseAndWithoutErrorPrefixParameter() throws IOException {
+        shouldPerformRequest(TEST_URL, HttpOpener.Method.GET, (o, u) -> o.setErrorPrefix(null),
+                null, null, WireMock.badRequest().withBody(RESPONSE_BODY), TEST_ERROR);
+    }
```

Comment:
There's no test coverage for `errorStream` yet. Wasn't able to come up with a way to trigger it.